components/service/glean/src/main/java/mozilla/components/service/glean/storages/GenericScalarStorageEngine.kt

```diff
@@ -105,13 +112,14 @@ internal abstract class GenericScalarStorageEngine<ScalarType> : StorageEngine {
 
             // Split the stored name in 2: we expect it to be in the format
             // store#metric.name
-            val (storeName, metricName) =
-                metricStoragePath.split('#', limit = 2)
+            val (storeName, metricName) = metricStoragePath.split('#', limit = 2)
+
             if (storeName.isEmpty()) {
                 continue
             }
 
-            val storeData = dataStores[Lifetime.User.ordinal].getOrPut(storeName) { mutableMapOf() }
+            val storeData = getDataStoreForLifetime(lifetime, storeName)
```

Comment:
I think this could just become `val storeData = dataStores[lifetime.ordinal].getOrPut(storeName) { mutableMapOf() }`: this way we can get rid of the new function, possibly removing the need for the suppression at the top of the class too.